### DISCUSSION TITLE
Selection of browser target binary and profile

### DISCUSSION THREAD

**Discussion by schenktobi** [aklinker1/vite-plugin-web-extension#186] [Q&A] [answered]
In the sourcecode for this package there is a mention, that we can pass a web-ext config to change for example the chromium or firefox binary, which will be used to install and run the extension in (https://github.com/mozilla/web-ext/blob/666886f40a967b515d43cf38fc9aec67ad744d89/src/program.js#L559). 

However after changing these Variables around and trying to get to load google-chrome with a specific profile by settings the binary of Google Chrome and the path to the profile on my mac book, when running `vite dev` the browser instance started still is another chromium instance with no profile used. Instead of my usual Chrome profiles being present there is only a default profile.

With that profile it's hard to test the extension because of missing logins etc.

I also logged the web-ext config in the sourcecode of this package and it seems to correctly set that, so I am really confused on how I can customize the browser launching behaviours

**Comment by schenktobi** [comment, marked as answer]
The solution was that despite the blob saying the flag is `chromium-binary` you need to use `chromiumBinary` in order to set a specific binary. I simply set a profile name and `createProfileIfMissing` to then keep the changes inside that profile like logins etc

**Reply by aklinker1** [reply]
For future reference, this is well documented:

https://vite-plugin-web-extension.aklinker1.io/guide/configure-browser-startup.html#available-options

**Reply by schenktobi** [reply]
well damn, thank you for answering so quick. I totally messed up my 10 cmd+f attemps then. I seem to be blind and oversaw that page. Maybe it helps someone else in the future. If you don't thinks so you can ofc delete the Discussion.

**Comment by relax-code-relax-life** [comment]
I encountered the same problem. And it is not working correctly.
I aspire to open the Chrome instance exactly as the user did, and it can even be opened by the current Chrome session.

So what should I do next?  I'm looking forward to someone's help and  thank you very much!

The following is my configuration:
```javascript
export default defineConfig({
  plugins: [
    webExtension({
      manifest: generateManifest,
      watchFilePaths: ["package.json", "manifest.json"],
      webExtConfig: {
        chromiumBinary: "/Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome",
        args: [
            "--user-data-dir=~/Library/Application\ Support/Google/Chrome/Default",
            '--profile-directory=~/Library/Application\ Support/Google/Chrome'
        ],
      }
    }),
  ],
});
```
